### PR TITLE
Training peaks api

### PR DESCRIPTION
**Pull request by sharoonarif**
### Work Items
* Backlog Item: #TFS15327
* Tasks:
  * #TFS15370

* This repo needs renaming! Not sure that I can do that 🤷‍♂️

### CODE REVIEW

File: TrainingPeaks/TrainingPeaksApi.cs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TrainingPeaks
+{
+	public class TrainingPeaksApi
+	{
+		private const string authCodeGrantType    = "authorization_code";
+		private const string refreshGrantType     = "refresh_token";
```
Comment (kapral) [inline TrainingPeaks/TrainingPeaksApi.cs]: it's inexcusable

File: TrainingPeaks/TrainingPeaksApi.cs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+		/// </summary>
+		/// <param name="apiParameters">The API parameters.</param>
+		public TrainingPeaksApi(ApiParameters apiParameters)
+		{
+			_apiParameters               = apiParameters ?? throw new ArgumentNullException(nameof(apiParameters));
+			_oAuthHttpClient.BaseAddress = apiParameters.OAuthUrl;
+			_apiHttpClient.BaseAddress   = apiParameters.ApiUrl;
+		}
+
+		public async Task<AthleteProfile> GetProfile(string accessToken)
```
Comment (kapral) [inline TrainingPeaks/TrainingPeaksApi.cs]: not consistent xml comments on public methods

File: TrainingPeaks/TrainingPeaksApi.cs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+			};
+		}
+
+		private async Task<TResult> GetApiResult<TResult>(HttpResponseMessage response)
+		{
+			var content = await response.Content.ReadAsStringAsync();
+			if (!response.IsSuccessStatusCode)
+				throw new ApiClientException(content);
+
+			return JsonConvert.DeserializeObject<TResult>(content);
```
Comment (kapral) [inline TrainingPeaks/TrainingPeaksApi.cs]: was there a reason of not doing `response.Content.ReadAsAsync<TResult>()`?

File: TrainingPeaks/TrainingPeaksApi.cs (hunk 2 above)
Comment (kapral) [inline TrainingPeaks/TrainingPeaksApi.cs]: i see it, ignore that

File: TrainingPeaks/TrainingPeaksApi.cs (hunk 3 above)
Comment (sharoonarif) [inline TrainingPeaks/TrainingPeaksApi.cs]: need nuget package for that extension `Microsoft.AspNet.WebApi.Client`, worth adding it?

**Pull request merged**